src/consensus/mod.rs

```diff
@@ -25,22 +45,38 @@ impl<'a, C, B> FastPaxos<'a, C, B> {
         size: usize,
         client: &'a mut C,
         broadcast: &'a mut B,
+        decision_tx: oneshot::Sender<Vec<Endpoint>>,
+        proposal_rx: oneshot::Receiver<Vec<Endpoint>>,
+        config_id: usize,
     ) -> FastPaxos<'a, C, B>
     where
         C: Client,
         B: Broadcast,
     {
         FastPaxos {
-            my_addr: my_addr.clone(),
             broadcast,
-            size,
-            paxos: Paxos::new(client, size, my_addr),
+            decision_tx,
+            config_id,
+            size: size,
+            my_addr: my_addr.clone(),
+            paxos: Paxos::new(client, size, my_addr, config_id),
+            decided: AtomicBool::new(false),
+            scheduled_paxos: None,
         }
     }
 
-    pub async fn propose(&self, proposal: Vec<Endpoint>) {
-        // TODO: register proposal with regular paxos instance
-        unimplemented!();
+    pub async fn propose(&self, proposal: Vec<Endpoint>) -> Result<()> {
+        let mut paxos_delay = Delay::new(Instant::now() + self.get_random_delay()).fuse();
+
+        // a hacky way to schedule a task after the delay
+        async {
```

Comment:
not sure I understand this hack, isnt it roughly the same thing?

Reply:
I wanted to have a way to schedule the classic round task in the background after a delay. I didn't know if there was a better way to do this, and it felt kinda hacky.

Reply:
isnt this just the same as the two awaits one after each other?

Reply:
Yeah you are right, it is. So how _would_ you schedule a background task?